### PR TITLE
Tracing fixes

### PR DESCRIPTION
**Pull request by marc-hb**
Various fixes and some comments. Main changes:

`dma-trace.c: remove misunderstood trace_flush() after FW ABI banner`
`trace: initialize DMA trace _after_ mailbox`

### CODE REVIEW

**Comment by marc-hb** [comment]
Two platforms were unavailable in https://sof-ci.01.org/sofpr/PR4327/build9313/devicetest/ otherwise everything is green.

The 2 checkpatch warnings in https://github.com/thesofproject/sof/pull/4327/checks?check_run_id=2789960845 are wrong:

- URL is too long
- 'ba' may be misspelled - perhaps 'by'? in 0xbe05ba30

**Review by lgirdwood — APPROVED** [review]

File: src/include/sof/trace/trace.h [hunk 1]
```diff
@@ -175,7 +175,7 @@ int trace_filter_update(const struct trace_filter *elem);
 	} log_entry = {						\
 		lvl,						\
 		comp_class,					\
-		params,						\
+		n_params,						\
```
Comment (ujfalusi) [inline src/include/sof/trace/trace.h:178]: While it makes the macro readable, it does not belong to this commit as per the subject and message

Comment (marc-hb) [reply src/include/sof/trace/trace.h:178]: Thanks @ujfalusi for the very thorough review! Technically you're right: this is not just a  comment change. In practice this macro parameter name is used only once in the macro definition, must be a compile-time constant and the macro itself is used in only one place in the entire code base right now. So the risk of conflict and possible consequences are either zero or extremely close to it and that's why I chose to save one extra commit for these 4 characters. As you've noticed the rest of the PR is carefully divided in 9 different commits with hopefully good commit messages.

Now if you insist... :-)

Comment (lgirdwood) [reply src/include/sof/trace/trace.h:178]: I'm fine - CI has already run. Lets merge.

Comment (ujfalusi) [reply src/include/sof/trace/trace.h:178]: @marc-hb, @lrgirdwo, I did not wanted to insist either.

This is a really nice cleanup series!
